src/webgl/p5.Shader.js

```diff
@@ -44,38 +77,28 @@ var p5 = require('../core/core');
  *
  * var mandel;
  * function setup() {
- *   var renderer = createCanvas(100, 100, WEBGL);
- *   mandel = new p5.Shader(renderer, vs, fs);
+ *   // create and initialize the shader
+ *   mandel = createShader(vs, fs);
  *   shader(mandel);
- *   mandel.setUniform('p', [-0.74364388703, 0.13182590421]);
  *   noStroke();
+ *
+ *   // 'p' is the center point of the Mandelbrot image
+ *   mandel.setUniform('p', [-0.74364388703, 0.13182590421]);
  * }
  *
  * function draw() {
+ *   // 'r' is the size of the image in Mandelbrot-space
  *   mandel.setUniform('r', 1.5 * exp(-6.5 * (1 + sin(millis() / 2000))));
  *   quad(-1, -1, 1, -1, 1, 1, -1, 1);
  * }
  * </code>
  * </div>
+ *
+ * @alt
+ * zooming Mandelbrot set. a colorful, infinitely detailed fractal.
  */
-p5.Shader = function(renderer, vertSrc, fragSrc) {
-  // TODO: adapt this to not take ids, but rather,
-  // to take the source for a vertex and fragment shader
-  // to enable custom shaders at some later date
-  this._renderer = renderer;
-  this._vertSrc = vertSrc;
-  this._fragSrc = fragSrc;
-  this._vertShader = -1;
-  this._fragShader = -1;
-  this._glProgram = 0;
-  this._loadedAttributes = false;
-  this.attributes = {};
-  this._loadedUniforms = false;
-  this.uniforms = {};
-  this._bound = false;
-  this.samplers = [];
-
-  return this;
+p5.prototype.createShader = function(vs, fs) {
```

Comment:
this looks great. the rest of this file only defines the Shader class, but no p5.prototype methods. I think it feels ok here but wonder if it belongs in material.js alongside loadShader? this file has more of the p5 public methods for working with shaders.